src/plot_api/plot_api.js

```diff
@@ -181,9 +176,9 @@ Plotly.plot = function(gd, data, layout, config) {
         var calcdata = gd.calcdata;
         var i, cd, trace;
 
-        Legend.draw(gd);
-        RangeSelector.draw(gd);
-        UpdateMenus.draw(gd);
+        Registry.getComponentMethod('legend', 'draw')(gd);
```

Comment:
Unlike traces and plot module, we can't simply loop over them in a given. Some components may require two draw calls per plot call, some component may not. Some components may have a calc step, some components may not etc.

So instead, we use a getter method to the component method on-demand. Note that if a given component isn't register, `Registry.getComponentMethod()` will return a `noop`.